File: libraries/cyclestreets-fragments/src/main/java/net/cyclestreets/ItineraryAndElevationFragment.java
package net.cyclestreets;

import android.os.Bundle;
import android.support.v4.app.ActivityCompat;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentTransaction;
import android.view.LayoutInflater;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;

import net.cyclestreets.fragments.R;

import static net.cyclestreets.util.MenuHelper.showMenuItem;

public class ItineraryAndElevationFragment extends Fragment {
  private Fragment lastFrag_;
  private Fragment itinerary_;
  private Fragment elevation_;

  @Override
  public void onCreate(Bundle savedInstanceState) {
    setRetainInstance(true);
    itinerary_ = new ItineraryFragment();
    elevation_ = new ElevationProfileFragment();

    super.onCreate(savedInstanceState);
  } // onCreate

  @Override
  public View onCreateView(final LayoutInflater inflater,
                           final ViewGroup container,
                           final Bundle savedInstanceState) {
    return inflater.inflate(R.layout.itinerary_and_elevation, container, false);
  } // onCreateView

  @Override
  public void onPause() {
    super.onPause();
  } // onPause

  @Override
  public void onResume() {
    super.onResume();
    showFrag(lastFrag_ != null ? lastFrag_ : itinerary_);
  } // onResume

  private void showFrag(Fragment frag) {
    FragmentManager fm = getChildFragmentManager();
    FragmentTransaction ft = fm.beginTransaction();

    if (lastFrag_ != null)
      ft.detach(lastFrag_);

    if (frag != null) {
      String tag = frag.getTag();
      if (fm.findFragmentByTag(tag) == null)
        ft.add(R.id.container, frag, frag.getClass().getSimpleName());
      else
        ft.attach(frag);
    } // if ...

    ft.commit();

    lastFrag_ = frag;

    ActivityCompat.invalidateOptionsMenu(getActivity());
  } // showFrag

  @Override
  public void onCreateOptionsMenu(final Menu menu, final MenuInflater inflater) {
    inflater.inflate(R.menu.itinerary_and_elevation_menu, menu);
    super.onCreateOptionsMenu(menu, inflater);
  } // onCreateOptionsMenu

  @Override
  public void onPrepareOptionsMenu(final Menu menu) {
    showMenuItem(menu, R.id.ic_menu_itinerary, itinerary_ != lastFrag_);
    showMenuItem(menu, R.id.ic_menu_elevation, elevation_ != lastFrag_);
    super.onPrepareOptionsMenu(menu);
  } // onPrepareOptionsMenu


  @Override
  public boolean onOptionsItemSelected(final MenuItem item) {
    if (super.onOptionsItemSelected(item))
      return true;

    final int menuId = item.getItemId();

    if (R.id.ic_menu_itinerary == menuId)
      showFrag(itinerary_);

    if (R.id.ic_menu_elevation == menuId)
      showFrag(elevation_);

    return true;
  } // onMenuItemSelected
} // ItineraryAndElevationFragment

File: libraries/cyclestreets-fragments/src/main/java/net/cyclestreets/itinerary/ItineraryAndElevationFragment.kt
package net.cyclestreets.itinerary

import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.Menu
import android.view.MenuInflater
import android.view.MenuItem
import android.view.View
import android.view.ViewGroup

import net.cyclestreets.fragments.R

import net.cyclestreets.util.MenuHelper.showMenuItem

class ItineraryAndElevationFragment : Fragment() {
    private var lastFrag: Fragment? = null
    private lateinit var itinerary: Fragment
    private lateinit var elevation: Fragment

    override fun onCreate(savedInstanceState: Bundle?) {
        setHasOptionsMenu(true)
        retainInstance = true
        itinerary = ItineraryFragment()
        elevation = ElevationProfileFragment()

        super.onCreate(savedInstanceState)
    }

    override fun onCreateView(inflater: LayoutInflater,
                              container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.itinerary_and_elevation, container, false)
    }

    override fun onResume() {
        super.onResume()
        showFrag(lastFrag ?: itinerary)
    }

    private fun showFrag(frag: Fragment) {
        val fm = childFragmentManager
        val ft = fm.beginTransaction()

        if (lastFrag != null)
            ft.detach(lastFrag)

        if (fm.findFragmentByTag(frag.tag) == null)
            ft.add(R.id.container, frag, frag.javaClass.simpleName)
        else
            ft.attach(frag)
        ft.commit()

        lastFrag = frag

        activity!!.invalidateOptionsMenu()
    }

    override fun onCreateOptionsMenu(menu: Menu?, inflater: MenuInflater) {
        inflater.inflate(R.menu.itinerary_and_elevation_menu, menu)
        super.onCreateOptionsMenu(menu, inflater)
    }

    override fun onPrepareOptionsMenu(menu: Menu) {
        showMenuItem(menu, R.id.ic_menu_itinerary, itinerary !== lastFrag)
        showMenuItem(menu, R.id.ic_menu_elevation, elevation !== lastFrag)
        super.onPrepareOptionsMenu(menu)
    }

    override fun onOptionsItemSelected(item: MenuItem?): Boolean {
        if (super.onOptionsItemSelected(item))
            return true

        val menuId = item!!.itemId

        if (R.id.ic_menu_itinerary == menuId)
            showFrag(itinerary)

        if (R.id.ic_menu_elevation == menuId)
            showFrag(elevation)

        return true
    }
}
